src/broker.ts

```diff
@@ -0,0 +1,100 @@
+import checkTypes = require("check-types");
+import traverson = require("traverson-promise");
+import JsonHalAdapter = require("traverson-hal");
+import q = require("q");
+import logger from "./logger";
+
+// register the traverson-hal plug-in for media type "application/hal+json"
+traverson.registerMediaType(JsonHalAdapter.mediaType, JsonHalAdapter);
+
+const pactURLPattern = "/pacts/provider/%s/latest";
+const pactURLPatternWithTag = "/pacts/provider/%s/latest/%s";
+
+export class Broker {
+	public static create(options: BrokerOptions = {}) {
+		// defaults
+		options.brokerUrl = options.brokerUrl || "";
+		options.provider = options.provider || "";
+		options.username = options.username || "";
+		options.password = options.password || "";
+		options.tags = options.tags || [];
+
+		checkTypes.assert.nonEmptyString(options.brokerUrl);
```

Comment:
Could we not use stronger typing here in the `BrokerOptions` to avoid checking all the types etc.?

Reply:
Will need to add runtime type checking eventually.